frontend/src/routes/Infrastructure/Clusters/ManagedClusters/ImportCluster/ImportCluster.tsx

```diff
@@ -20,58 +21,148 @@ import {
     AcmButton,
     AcmForm,
     AcmLabelsInput,
-    AcmPage,
     AcmPageContent,
     AcmPageHeader,
     AcmSelect,
     AcmSubmit,
     AcmTextArea,
     AcmTextInput,
 } from '@open-cluster-management/ui-components'
-import { ActionGroup, Button, Label, PageSection, SelectOption, Text } from '@patternfly/react-core'
+import {
+    ActionGroup,
+    Button,
+    Drawer,
+    DrawerColorVariant,
+    DrawerContent,
+    DrawerContentBody,
+    DrawerPanelContent,
+    FormGroup,
+    Label,
+    PageSection,
+    SelectOption,
+    Text,
+    Page,
+    Switch,
+} from '@patternfly/react-core'
 import { CheckCircleIcon } from '@patternfly/react-icons'
 import '@patternfly/react-styles/css/components/CodeEditor/code-editor.css'
-import { Fragment, useContext, useState } from 'react'
+import { Fragment, useContext, useRef, useState, useEffect } from 'react'
+import useResizeObserver from '@react-hook/resize-observer'
 import { useTranslation } from 'react-i18next'
 import { Link, useHistory } from 'react-router-dom'
 import { DOC_LINKS } from '../../../../../lib/doc-util'
 import { NavigationPath } from '../../../../../NavigationPath'
 import { useCanJoinClusterSets, useMustJoinClusterSet } from '../../ClusterSets/components/useCanJoinClusterSets'
 import { ImportCommand, pollImportYamlSecret } from '../components/ImportCommand'
+import { keyBy } from 'lodash'
+import { SyncEditor } from '../../../../../components/SyncEditor/SyncEditor'
+import { SyncDiff, SyncDiffType } from '../../../../../components/SyncEditor/SyncDiff'
+import schema from './schema.json'
+
+const minWizardSize = 1000
+const defaultPanelSize = 600
 
 export default function ImportClusterPage() {
     const { t } = useTranslation(['cluster'])
+    const pageRef = useRef(null)
+    const [drawerExpanded, setDrawerExpanded] = useState(localStorage.getItem('import-cluster-yaml') === 'true')
```

Comment:
Looks like the drawer was added to the import cluster page instead of the common code.
DataForm already has code for showing a yaml view.

Reply:
that's great but ImportCluster doesn't use DataForm